### PR TITLE
Cleanup get user and facility function

### PR DESCRIPTION
**Pull request by EdisonTing**
Cleanup/rewrite getUserAndFacility function to return empty object when there's no user and facility

### CODE REVIEW

File: prototypes/rpcRunner.js [hunk 1]
```diff
@@ -127,21 +127,37 @@ RPCRunner.prototype.initializeUser = function(userId, context) {
...
-    vals["facilityId"] = parseInt(pieces[2]);
-    vals["facilityName"] = pieces[3];
-    if ((pieces.length === 5) && (pieces[4] !== ""))
-        vals["facilityStationNumber"] = pieces[4];
+    var res = this.db.function({ function: "GETUNF^RPCRUN" });
+
+    if (!_.isUndefined(res.result) && _.isString(res.result)) {
+        results = res.result.split(":");
+        if (results.length > 1) {
+            _.each(results, function(value, index) {
```
Comment (mfuroyama) [inline prototypes/rpcRunner.js]: It's all good...change 'em!

Just a thought, but I wonder if you can use lodash `_.transform()` here? Might make things simpler?

Comment (EdisonTing) [inline prototypes/rpcRunner.js]: hm.. how?

Comment (mfuroyama) [inline prototypes/rpcRunner.js]:
```Javascript
if (results.length > 1) {
    _.transform(results, (result, value, index) => {
        const key = keys[index];
        results[key.name] = (key.type === 'int' ? parseInt(value, 10) : value);
    }, vals);
}
```

Actually, after looking at the implementation, it doesn't matter; it's about the same either way... 😄

Comment (EdisonTing) [inline prototypes/rpcRunner.js]: cool. good to know!

**Pull request merged**